src/button-group/button-group.js

```diff
@@ -127,36 +127,15 @@ export default class ButtonGroup extends React.Component<PropsT> {
                       }
 
                       if (shape !== SHAPE.default) {
-                        if (index === 0) {
-                          return {};
-                        }
                         return {
-                          marginLeft: $theme.sizing.scale300,
+                          marginLeft: $theme.sizing.scale100,
+                          marginRight: $theme.sizing.scale100,
                         };
                       }
 
-                      // left most button
-                      if (index === 0) {
-                        return {
-                          borderTopRightRadius: 0,
-                          borderBottomRightRadius: 0,
-                        };
-                      }
-                      // right most button
-                      if (index === children.length - 1) {
-                        return {
-                          borderTopLeftRadius: 0,
-                          borderBottomLeftRadius: 0,
-                          marginLeft: '1px',
-                        };
-                      }
-                      // inner button(s)
                       return {
-                        borderTopRightRadius: 0,
```

Comment:
Why remove the additional styles? I believe this is here to support older rounded corner buttons

Reply:
I think the current also doesn't work for RTL. We probably could add even more conditions but it feels strange adding all this logic trying to support possible override changes? Also, it's been a while since we had round buttons.

Reply:
If people wish to have round buttons in a group with inner squared button, they should imo implement their own button group, not trying to rely on our styles that are based around differently shaped buttons.